Cargo.toml

```diff
@@ -33,6 +33,8 @@ diesel1 = { default-features = false, optional = true, package = "diesel", versi
 diesel2 = { default-features = false, optional = true, package = "diesel", version = "2.0" }
 num-traits = { default-features = false, features = ["i128"], version = "0.2" }
 postgres = { default-features = false, optional = true, version = "0.19" }
+proptest = { default-features = false, optional = true, features = ["std"], version = "1.0" }
```

Comment:
Is `std` required? You could potentially add this to the `std` feature of Rust Decimal - i.e. `"proptest?/std"`

Reply:
Unfortunately without it I get this error.

```
error[E0207]: the type parameter `T` is not constrained by the impl trait, self type, or predicates
   --> /Users/cardoe/.cargo/registry/src/index.crates.io-6f17d22bba15001f/proptest-1.1.0/src/collection.rs:219:23
    |
219 |     [] fn VecToDeque[<T : fmt::Debug>](vec: Vec<T>) -> VecDeque<T> {
    |                       ^ unconstrained type parameter
```

I'm compiling rust-decimal with `cargo test --no-default-features --features proptest`

It's possible I'm missing something else or proptest has an issue?

Reply:
So proptest requires an unstable compiler to work with no-std per their docs at https://altsysrq.github.io/proptest-book/proptest/no-std.html and they have an opened issue at proptest-rs/proptest#47 for resolving this. So can we leave the dependency on proptest with std?

Reply:
Ok, interesting. While we can force `std` within the rust decimal feature it certainly highlights a flaw in the ecosystem - which isn't unique to `proptest`. 

Rust Decimal is really a small data type within the rust ecosystem providing not much more than a type to calculate decimal numbers. To provide support for various frameworks it is exposing features which provide compatibility. While exposing features for compatibility isn't a fundamental problem, having rust decimal pull in the dependencies etc of the peering library is. _Ideally_ if rust decimal was to provide the compatibility layer, it'd only need to pull in the smallest surface area of the library as possible. For `proptest` it'd really just be the `Arbitrary` trait and dependent types. That, or the inverse relationship should be exposed (i.e. `proptest` expose a rust decimal feature). The issue with the current approach is that we force `proptest` support to be `std` - whereas it shouldn't need to be. Rust Decimal certainly doesn't require `std` so we shouldn't really force a downstream dependency to be that either.

Again, this isn't unique to this PR/library - it's just a comment on the eco-system. There is already a precedent for rust decimal to provide the compatibility layer however it is perhaps something that I may need to rethink for v2. It's not an easy thing to solve as the intention is not to have a "stand off" with other libraries, however rather provide the most intuitive and flexible solution for the end user. Anyway!

Reply:
I think it would be great if there was a compatible Arbitrary trait between proptest, rust-fuzz and other testing libraries. None of them are compatible with each other unfortunately.

I totally agree with you about this issue in the eco-system. There's a couple of other areas in Rust that are wart-y in my opinion as well. Like MSRV support.